enhancements/20240820-mtu-and-network-information-propagation.md

```diff
@@ -0,0 +1,318 @@
+
+### Related Issues
+
+The URL For the related enhancement issues in Harvester repo.
+
+#### Issue Type 1: MTU is not synced to NAD
+
+[BUG The bridge MTU is not changed](https://github.com/harvester/harvester/issues/4353)
+
+MTU of the uplink is set as 9000; MTU of the bridge (e.g. storage-br) is still 1500 but the bond's is 9000
+
+Closed, via PR [Sync MTU value to NAD](https://github.com/harvester/network-controller-harvester/pull/87)
+
+#### Issue Type 2: Change MTU on the Fly
+
+[Question Should updated MTU apply on existed VM network? #4389](https://github.com/harvester/harvester/issues/4389)
+
+Closed, expected to be solved on issue #4355.
+
+[The MTU value does not propagate to all the interfaces if MTU is configured after the VlanConfig creation #4752](https://github.com/harvester/harvester/issues/4752)
+
+Closed, expected to be solved on issue #4355.
+
+#### Issue Type 3: Move MTU to Clusternetwork
+
+[ENHANCEMENT Support configure MTU in the cluster network #4355](https://github.com/harvester/harvester/issues/4355)
+
+#### JIRA ticket: Question about Jumbo frames in Harvester
+
+[Question about Jumbo frames in Harvester](https://jira.suse.com/browse/SURE-7171)
+
+>What sense does it make to have Jumbo frames (MTU 9000) on Harvester physical interfaces?
+When jumbo frames are configured, the CNI is not configured accordingly. The MTU of all pod interfaces is 1450 (1500 minus vxlan overhead). Even when the storage network is configured, the MTU of the virtual storage adapter is 1500. As a result, all relevant traffic is using MTU 1500. Only node traffic, e.g. image pulls can (in theory) profit from MTU 9000, but only if the registry is also using jumbo frames. E.g. VM images are downloaded from inside a pod and therefore use MTU 1500. Please describe a scenario where (and how) jumbo frames are useful and increase the performance!
+How about the possibility of using jumbo frames in guest clusters?
+
+:::note
+
+We did not give a formal answer to this yet, it is complex and needs testing.
+
+:::
+
+## Motivation
+
+In the context of Harvester, user may ask:
+
+(1) Can a network with MTU 9000 benefit Harvester [Storage Network](https://docs.harvesterhci.io/v1.3/advanced/storagenetwork), specifically Longhorn?
+
+A detailed analysis is needed to identify the L2 packet characters of Longhorn data replication. 
+
+It depends how LH copies the trunks of data from its TCP/IP connection in most cases. 1K, 4K or 1M? If a big part of the packets are under 1500, then MTU 9000 does not bring big improvements. Meanwhile, sending 1000 packets in 0.1 second or 0.001 second also makes things different. When the traffic is high and there are potential network related pressures, the higher MTU may help.
+
+This applies to Kubernetes management network as well.
```

Comment:
Is this line line relevant here ? network config on a mgmt cluster cannot be modified by user.In that case I believe mgmt network functions with default MTU.

Reply:
We allow to set a different MTU for management network in PXE mode.
https://docs.harvesterhci.io/v1.3/install/harvester-configuration#installmanagement_interface

I doubt/am not sure: the MTU 9000 can bring noticeable improvements.